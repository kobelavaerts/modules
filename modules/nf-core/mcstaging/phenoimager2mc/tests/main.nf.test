nextflow_process {

    name "Test Process MCSTAGING_PHENOIMAGER2MC"
    script "../main.nf"
    process "MCSTAGING_PHENOIMAGER2MC"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "mcstaging"
    tag "mcstaging/phenoimager2mc"
    // tag "untar"


    // setup {
    //     run("UNTAR") {
    //         script "modules/nf-core/untar/main.nf"
    //         process {
    //             """
    //             input[0] = Channel.of([ [id: "test"], file("https://github.com/kobelavaerts/mock-test-data/raw/refs/heads/main/tiles.tar.gz", checkIfExists: true) ])
    //             """
    //         }
    //     }
    // }

    test("phenoimager2mc - tif") {
        when {
            process {
                """
                // define inputs of the process here. Example:
                input[0] = [
                    [ id:'test' ],
                    [
                        file(params.modules_testdata_base_path + 'imaging/staging/phenoimager/tile1.tif', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'imaging/staging/phenoimager/tile2.tif', checkIfExists: true)
                    ]
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                      file(process.out.tif[0][1]).name,
                      process.out.versions).match() }
            )
        }
    }


test("phenoimager2mc - tif - stub") {

    options "-stub"
        when {
            process {
                """
                // define inputs of the process here. Example:
                input[0] = [
                    [ id:'test' ],
                    [
                        file(params.modules_testdata_base_path + 'imaging/staging/phenoimager/tile1.tif', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'imaging/staging/phenoimager/tile2.tif', checkIfExists: true)
                    ]
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                      file(process.out.tif[0][1]).name,
                      process.out.versions).match() }
            )
        }
    }

    // test("phenoimager2mc - tif_folder") {
    //     when {
    //         process {
    //             """
    //             // define inputs of the process here. Example:
    //             input[0]= UNTAR.out.untar

    //             """
    //         }
    //     }
    //     then {
    //         assertAll (
    //             { assert process.success },
    //             { assert snapshot(
    //                   file(process.out.tif[0][1]).name,
    //                   process.out.versions).match() }
    //         )
    //     }
    // }

// tmpdir = file("${projectDir}/tmpdir", type: "dir")
// tmpdir.mkdir()
// tif_file_1 = file(params.modules_testdata_base_path + 'imaging/staging/phenoimager/tile1.tif', checkIfExists: true)
// tif_file_2 = file(params.modules_testdata_base_path + 'imaging/staging/phenoimager/tile2.tif', checkIfExists: true)
// tif_file_1.copyTo(tmpdir)
// tif_file_2.copyTo(tmpdir)


    test("phenoimager2mc - tif_folder_cp") {
        when {
            process {
                """
                tmpdir = file("tmpdir", type: 'dir')
                tmpdir.mkdir()
                tif_file_1 = file(params.modules_testdata_base_path + 'imaging/staging/phenoimager/tile1.tif', checkIfExists: true)
                tif_file_2 = file(params.modules_testdata_base_path + 'imaging/staging/phenoimager/tile2.tif', checkIfExists: true)
                tif_file_1.copyTo(tmpdir)
                tif_file_2.copyTo(tmpdir)
                // define inputs of the process here. Example:
                input[0] = [
                    [ id:'test' ],
                    [
                        file(tmpdir, type:'dir')
                    ]
                ]


                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(
                      file(process.out.tif[0][1]).name,
                      process.out.versions).match() }
            )
        }
    }
}
