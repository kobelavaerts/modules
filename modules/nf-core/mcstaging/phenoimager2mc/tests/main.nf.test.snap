{
    "phenoimager2mc - tif - stub": {
        "content": [
            "test.tif",
            [
                "versions.yml:md5,30d4a7af53d372a0700d388471500532"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-28T13:59:30.97214661"
    },
    "phenoimager2mc - tif": {
        "content": [
            "test.tif",
            [
                "versions.yml:md5,30d4a7af53d372a0700d388471500532"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T13:17:56.890612618"
    },
    "phenoimager2mc - tif_folder": {
        "content": [
            "test.tif",
            [
                "versions.yml:md5,30d4a7af53d372a0700d388471500532"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-28T13:43:34.891677783"
    },
    "phenoimager2mc - tif_folder_cp": {
        "content": [
            "test.tif",
            [
                "versions.yml:md5,30d4a7af53d372a0700d388471500532"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-28T15:39:37.135693287"
    }
}